mempool/src/pool.rs

```diff
@@ -236,9 +220,49 @@ where
             clock,
             memory_usage_estimator,
             events_controller: Default::default(),
+            receiver,
         }
     }
 
+    pub async fn subscribe_to_chainstate_events(
```

Comment:
Is `lib.rs` supposed to be able to call this function? Why does it have access to it?

Same story.

Reply:
I made `pool` a private submodule, akin to detail. I hope this addresses the issue.